src/ligero/mod.rs

```diff
@@ -21,5 +23,126 @@ pub struct LigeroParameters {
     /// The number of quadratic constraints written in each row. Also "QR".
     pub quadratic_constraints_per_row: usize,
     /// The size of each row, in terms of number of field elements. Also "BLOCK".
+    /// There is some confusion in the specification between this and NCOL.
     pub row_size: usize,
+    /// The total size of a tableau row.
+    /// There is some confusion in the specification between this and BLOCK.
+    pub num_columns: u128,
+}
+
+impl LigeroParameters {
+    /// DBLOCK, 2 * BLOCK - 1
+    pub fn dblock(&self) -> usize {
+        self.row_size * 2 - 1
+    }
+}
+
+/// The extend method, as defined in [2.2.1][1]. We interpolate a polynomial of degree `|nodes|`
```

Comment:
The degree of a polynomial will be one less than the number of coordinates needed to specify it, or lower.
```suggestion
/// The extend method, as defined in [2.2.1][1]. We interpolate a polynomial of degree at most `|nodes| - 1`
```